Panic during processing of empty compressed payloads

Happened once, can't reproduce.
```
panic: interface conversion: payload.Payload is *payload.NullPayload, not *payload.Version

goroutine 59 [running]:
github.com/nspcc-dev/neo-go/pkg/network.(*Server).handleMessage(0xc0002228c0, 0xec12a0, 0xc00027c090, 0xc00035ef18, 0x0, 0x0)
        /neo-go/pkg/network/server.go:757 +0xbb6
github.com/nspcc-dev/neo-go/pkg/network.(*TCPPeer).handleConn(0xc00027c090)
        /neo-go/pkg/network/tcp_peer.go:162 +0x2b6
created by github.com/nspcc-dev/neo-go/pkg/network.(*TCPTransport).Accept
        /neo-go/pkg/network/tcp_transport.go:66 +0x446
```
The problem is that later in code certain commands assumed to have payload of the specified type. https://github.com/nspcc-dev/neo-go/blob/master/pkg/network/message.go#L103
We need to take into account which commands can have empty payload and return error for others.

--- a/pkg/network/message.go
+++ b/pkg/network/message.go
@@ -106,7 +106,12 @@ func (m *Message) Decode(br *io.BinReader) error {
 	// check the length first in order not to allocate memory
 	// for an empty compressed payload
 	if l == 0 {
-		m.Payload = payload.NewNullPayload()
+		switch m.Command {
+		case CMDFilterClear, CMDGetAddr, CMDMempool:
+			m.Payload = payload.NewNullPayload()
+		default:
+			return errors.New("unexpected empty payload")
+		}
 		return nil
 	}
 	if l > PayloadMaxSize {

--- a/pkg/network/payload/payload.go
+++ b/pkg/network/payload/payload.go
@@ -12,12 +12,12 @@ type NullPayload struct {
 }
 
 // NewNullPayload returns zero-sized stub payload.
-func NewNullPayload() *NullPayload {
-	return &NullPayload{}
+func NewNullPayload() NullPayload {
+	return NullPayload{}
 }
 
 // DecodeBinary implements Serializable interface.
-func (p *NullPayload) DecodeBinary(r *io.BinReader) {}
+func (p NullPayload) DecodeBinary(r *io.BinReader) {}
 
 // EncodeBinary implements Serializable interface.
-func (p *NullPayload) EncodeBinary(w *io.BinWriter) {}
+func (p NullPayload) EncodeBinary(w *io.BinWriter) {}

